plugins/content/local/writer.go

```diff
@@ -94,7 +95,10 @@ func (w *writer) Commit(ctx context.Context, size int64, expected digest.Digest,
 		return fmt.Errorf("cannot commit on closed writer: %w", errdefs.ErrFailedPrecondition)
 	}
 
-	if err := fp.Sync(); err != nil {
+	// fp.Sync() may take serval minutes to write in-memory big data to local storage.
+	// Use Flush() method instead to accelerate it.
+	b := bufio.NewWriter(fp)
+	if err := b.Flush(); err != nil {
```

Comment:
From what I read in `bufio` source code, this code will just skip the flush:

https://github.com/golang/go/blob/097b7162adeab8aad0095303aff8a045bbbfa6e0/src/bufio/bufio.go#L631-L638

```golang
// Flush writes any buffered data to the underlying [io.Writer].
func (b *Writer) Flush() error {
	if b.err != nil {
		return b.err
	}
	if b.n == 0 {
		return nil
	}
```

as `n = 0` right after `NewWriter` call.

Reply:
> From what I read in `bufio` source code, this code will just skip the flush:
> 
> https://github.com/golang/go/blob/097b7162adeab8aad0095303aff8a045bbbfa6e0/src/bufio/bufio.go#L631-L638
> 
> ```go
> // Flush writes any buffered data to the underlying [io.Writer].
> func (b *Writer) Flush() error {
> 	if b.err != nil {
> 		return b.err
> 	}
> 	if b.n == 0 {
> 		return nil
> 	}
> ```
> 
> as `n = 0` right after `NewWriter` call.

Yes, I might have made a mistake. I need to create a bufio Writer interface before io.Copy, and then flush it. I will update my PR soon.